accumulator_updater/programs/accumulator_updater/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+mod macros;
+
+use anchor_lang::{
+    prelude::*,
+    solana_program::sysvar::{
+        self,
+        instructions::get_instruction_relative,
+    },
+    system_program::{
+        self,
+        CreateAccount,
+    },
+};
+
+declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+
+#[program]
+pub mod accumulator_updater {
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let whitelist = &mut ctx.accounts.whitelist;
+        whitelist.bump = *ctx.bumps.get("whitelist").unwrap();
+        Ok(())
+    }
+
+    //TODO: add authorization mechanism for this
+    pub fn add_allowed_program(
+        ctx: Context<AddAllowedProgram>,
+        allowed_program: Pubkey,
+    ) -> Result<()> {
+        let whitelist = &mut ctx.accounts.whitelist;
+        require_keys_neq!(allowed_program, Pubkey::default());
```

Comment:
it's not really necessary since i don't think it's possible to have a program with this Pubkey anyways. it was mostly just a sanitation check to avoid populating the whitelist with junk